kernel/io.cc

```diff
@@ -464,7 +436,7 @@ void format_emit_long_long(std::string &result, std::string_view spec, int *dyna
 {
 	if (spec == "%d") {
 		// Format checking will have guaranteed num_dynamic_ints == 0.
-		result += std::to_string(arg);
+		result += std::to_string(static_cast<int>(arg));
```

Comment:
Why are we casting to a regular int in `format_emit_long_long`?

Reply:
I suppose it would be clearer if the function was called `format_emit_signed_integer()`. I could write a PR to make that change (and a corresponding change in the `ConversionSpecifier` enum), and similar for the other functions, if you think that's helpful.